pylint/config/arguments_manager.py

```diff
@@ -22,7 +22,7 @@ def __init__(self) -> None:
         self.namespace = argparse.Namespace()
         """Namespace for all options."""
 
-        self._arg_parser = argparse.ArgumentParser(prog="pylint")
+        self._arg_parser = argparse.ArgumentParser(prog="pylint", allow_abbrev=False)
```

Comment:
@Pierre-Sassoulas This is a crucial parameter. Otherwise `--ignore` can be used for `--ignore-rgx`, `--ignore-classes`, etc.

Should have done this in the first PR that got merged. Sorry!

Reply:
Yeah we're going to need it until the end of time seeing how many args we have 😄

Reply:
I didn't know it was turned off be default. Oh well...

Reply:
I just learned that I can probably type only half of the args in most of my python utilities, I'm going to save so much time 😄